Zend/zend_vm_def.h

```diff
@@ -5327,32 +5327,36 @@ ZEND_VM_HANDLER(73, ZEND_INCLUDE_OR_EVAL, CONST|TMPVAR|CV, ANY, EVAL)
 	ZEND_VM_CONTINUE();
 }
 
-ZEND_VM_HANDLER(74, ZEND_UNSET_VAR, CONST|TMPVAR|CV, UNUSED, VAR_FETCH|ISSET)
+ZEND_VM_HANDLER(196, ZEND_UNSET_CV, CV, UNUSED)
 {
 	USE_OPLINE
-	zval tmp, *varname;
-	HashTable *target_symbol_table;
-	zend_free_op free_op1;
+	zval *var = EX_VAR(opline->op1.var);
 
-	SAVE_OPLINE();
-	if (OP1_TYPE == IS_CV &&
-	    (opline->extended_value & ZEND_QUICK_SET)) {
-		zval *var = EX_VAR(opline->op1.var);
-
-		if (Z_REFCOUNTED_P(var)) {
-			zend_refcounted *garbage = Z_COUNTED_P(var);
+	if (Z_REFCOUNTED_P(var)) {
+		zend_refcounted *garbage = Z_COUNTED_P(var);
 
-			ZVAL_UNDEF(var);
-			if (!--GC_REFCOUNT(garbage)) {
-				zval_dtor_func(garbage);
-			} else {
-				gc_check_possible_root(garbage);
-			}
+		ZVAL_UNDEF(var);
+		if (!--GC_REFCOUNT(garbage)) {
+			SAVE_OPLINE();
+			zval_dtor_func(garbage);
+			ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 		} else {
-			ZVAL_UNDEF(var);
+			gc_check_possible_root(garbage);
```

Comment:
This could trigger gc and throw, so it should also use the CHECK_EXCEPTION variant.